great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -0,0 +1,447 @@
+        ] = project_config.anonymous_usage_statistics
+        context_config_usage_stats: Optional[
+            AnonymizedUsageStatisticsConfig
+        ] = self.config.anonymous_usage_statistics
+
+        if (
+            project_config_usage_stats.enabled is False  # type: ignore[union-attr]
+            or context_config_usage_stats.enabled is False  # type: ignore[union-attr]
+        ):
+            return False
+
+        if project_config_usage_stats.explicit_id is False:  # type: ignore[union-attr]
+            return True
+
+        if project_config_usage_stats == context_config_usage_stats:
+            return False
+
+        if project_config_usage_stats is None or context_config_usage_stats is None:
+            return True
+
+        # If the data_context_id differs and that difference is not a result of a global override, a sync is necessary.
+        global_data_context_id: Optional[str] = self._get_data_context_id_override()
+        if (
+            project_config_usage_stats.data_context_id
+            != context_config_usage_stats.data_context_id
+            and context_config_usage_stats.data_context_id != global_data_context_id
+        ):
+            return True
+
+        # If the usage_statistics_url differs and that difference is not a result of a global override, a sync is necessary.
+        global_usage_stats_url: Optional[str] = self._get_usage_stats_url_override()
+        if (
+            project_config_usage_stats.usage_statistics_url
+            != context_config_usage_stats.usage_statistics_url
+            and context_config_usage_stats.usage_statistics_url
+            != global_usage_stats_url
+        ):
+            return True
+
+        return False
+
+    @classmethod
+    def create(
+        cls,
+        project_root_dir: Optional[str] = None,
+        usage_statistics_enabled: bool = True,
+        runtime_environment: Optional[dict] = None,
+    ) -> SerializableDataContext:
+        """
+        Build a new great_expectations directory and DataContext object in the provided project_root_dir.
+
+        `create` will create a new "great_expectations" directory in the provided folder, provided one does not
+        already exist. Then, it will initialize a new DataContext in that folder and write the resulting config.
+
+        --Public API--
+
+        --Documentation--
+            https://docs.greatexpectations.io/docs/terms/data_context
+
+        Args:
+            project_root_dir: path to the root directory in which to create a new great_expectations directory
+            usage_statistics_enabled: boolean directive specifying whether or not to gather usage statistics
+            runtime_environment: a dictionary of config variables that override both those set in
+                config_variables.yml and the environment
+
+        Returns:
+            DataContext
+        """
+
+        if not os.path.isdir(project_root_dir):  # type: ignore[arg-type]
```

Comment:
Since this is a net new module can we `pathlib` and remove `os`?
If compatibility is a concern we can just call `str(my_path_obj)` on the path object when passing it.

Reply:
I'd be happy to but would you mind if I did this `os` replacement work in a subsequent PR? Can add to the context epic I'm working in but I'd like to focus on getting this change in before it gets stale.

Will defer to you here.

Reply:
Sounds good to me.